src/main/java/com/github/arix3767/user/service/UserService.java

```diff
@@ -0,0 +1,48 @@
+package com.github.arix3767.user.service;
+
+import com.github.arix3767.enums.Messages;
+import com.github.arix3767.exception.CustomerAlreadyExistsException;
+import com.github.arix3767.exception.InvalidEmailException;
+import com.github.arix3767.exception.MissingDataException;
+import com.github.arix3767.exception.UserNotFoundException;
+import com.github.arix3767.user.converters.UserDtoToUserConverter;
+import com.github.arix3767.user.converters.UserToUserDtoConverter;
+import com.github.arix3767.user.dto.UserDto;
+import com.github.arix3767.user.entity.User;
+import com.github.arix3767.user.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+    private UserRepository userRepository;
+
+    public UserDto addUser(UserDto userDto) {
+        if (userDto.getEmail() == null || userDto.getPassword() == null) {
+            throw new MissingDataException();
+        }
+        if (userRepository.existByEmail(userDto.getEmail())) {
+            throw new CustomerAlreadyExistsException();
+        }
+        User user = UserDtoToUserConverter.INSTANCE.convert(userDto);
+        userRepository.save(user);
+        return UserToUserDtoConverter.INSTANCE.convert(user);
+    }
+
+    public String editUser(long id, UserDto newUserData) {
+        if (newUserData.getEmail() == null) {
+            throw new InvalidEmailException();
+        }
+        User user = userRepository.findById(id)
```

Comment:
Rezygnujemy z wyjątku na rzecz rezultatu. Flow:

1. znajdź użytkownika
2. jeśli nie, zwróć odpowiedni  rezultat
3. jeśli tak, edytuj dane
4. zapisz dane do repo
5. skonwertuj do DTO to, co zwróciło reo